### ISSUE TITLE
WGAN-GP Keras tutorial is slow to start the first epoch #48356

### ISSUE THREAD

**Issue by romanovzky** [keras-team/keras#14580]
Hi there, I have opened the same issue in the [TensorFlow github](https://github.com/tensorflow/tensorflow/issues/48356), as I am not sure where it fits better, but here it goes:

**Describe the current behavior**

When I run the  [WGAN-GP tutorial from the Keras website](https://keras.io/examples/generative/wgan_gp/), I notice that the first epoch takes a really long time to start in both CPU and GPU, up to a minute.

**Describe the expected behavior**

The first epoch should start promptly, as it happens with the other tutorials. For example the [VAE](https://keras.io/examples/generative/vae/) and traditional [DCGAN](https://keras.io/examples/generative/dcgan_overriding_train_step/)

**Standalone code to reproduce the issue**

The [WGAN-GP colab](https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/generative/ipynb/wgan_gp.ipynb). Choose a runtime accelerator and click on run all. The first epoch will always have a delay when starting training.

The [VAE colab](https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/generative/ipynb/vae.ipynb) and [DCGAN colab](https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/generative/ipynb/dcgan_overriding_train_step.ipynb) do not exhibit this regression.

**Other info**

I can reproduce the same performance issue on my local computer, so this is not a colab issue.

**Comment by qlzh727** [comment]
@AakashKumarNain who is the original author the notebook. https://github.com/keras-team/keras-io/pull/15.

**Comment by AakashKumarNain** [comment]
This is weird. Although there is nothing special about the model here, so model is definitely the problem. The only thing that can be a "bit" slow is the `gradient_penalty()` but I doubt that would add so much of training time. I would take a look at this. Thanks for reporting.

**Comment by romanovzky** [comment]
No worries, I'd be glad to have this fixed as I've been tinkering a lot more with custom architectures with explicite gradient tapes and this is a bit of a paper-cut. I'm happy to further debug if you need me check anything.

**Comment by AakashKumarNain** [comment]
Can you try to using the TF debugger and see which op is taking time?

**Comment by AakashKumarNain** [comment]
@romanovzky did you try using the TF debugger?

**Comment by romanovzky** [comment]
Hi, I'm sorry but I have a lot on my hands right now and my current 
on-going projects do not involve these type of architectures. Might be 
able to pick it up again, but not in the next few weeks.

**Comment by AakashKumarNain** [comment]
No worries.
